vessim/core.py

```diff
@@ -1,76 +1,28 @@
-            index marking the time, and each column should represent a different zone
-            containing the data. The name of the zone is equal to the column name.
-            Note that while interpolation is possible when retrieving forecasts, the
-            actual data between timestamps is computed using either `ffill` or `bfill`.
-            If you wish a different behavior, you have to change your actual data
-            beforehand (e.g. by resampling into a different frequency).
-
-        forecast: An optional time-series dataset representing forecasted values. The
-            data should contain two datetime-like indices. One is the
-            `Request Timestamp`, marking the time when the forecast was made. One is the
-            `Forecast Timestamp`, indicating the time the forecast is made for.
-
-            - If data does not include a `Request Timestamp`, it is treated as a static
-              forecast that does not change over time.
-
-            - If `forecast` is not provided, predictions are derived from the actual
-              data when requesting forecasts (actual data is treated as static forecast).
-
-            - When using a non-static forecast, you have to be aware that all the
-              zones/column names also have to appear in the actual dataframe because some
-              actual values are used for interpolation.
-
-        fill_method: Either `ffill` or `bfill`. Determines how actual data is acquired in
-            between timestamps. Default is `ffill`.
-
-    Example:
-        >>> actual_data = [
-        ...    ["2020-01-01T00:00:00", 100, 800],
-        ...    ["2020-01-01T00:30:00", 110, 850],
-        ...    ["2020-01-01T01:00:00", 105, 820],
-        ...    ["2020-01-01T01:30:00", 108, 840],
-        ... ]
-        >>> actual = pd.DataFrame(
-        ...    actual_data, columns=["timestamp", "zone_a", "zone_b"]
-        ... )
-        >>> actual.set_index(["timestamp"], inplace=True)
-
-        >>> forecast_data = [
-        ...    ["2020-01-01T00:00:00", "2020-01-01T00:30:00", 115, 850],
-        ...    ["2020-01-01T00:00:00", "2020-01-01T01:00:00", 110, 870],
-        ...    ["2020-01-01T00:00:00", "2020-01-01T01:30:00", 110, 860],
-        ...    ["2020-01-01T00:00:00", "2020-01-01T02:00:00", 120, 830],
-        ...    ["2020-01-01T01:00:00", "2020-01-01T01:30:00", 115, 840],
-        ...    ["2020-01-01T01:00:00", "2020-01-01T02:00:00", 110, 830],
-        ... ]
-        >>> forecast = pd.DataFrame(
-        ...    forecast_data, columns=["req_time", "forecast_time", "zone_a", "zone_b"]
-        ... )
-        >>> forecast.set_index(["req_time", "forecast_time"], inplace=True)
-
-        >>> time_series = TimeSeriesApi(actual, forecast)
-    """
+from vessim.data import _load_dataset, _convert_to_datetime, DatetimeLike
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+class Api(ABC):
```

Comment:
yes I was also not very happy with the name, especially since the word API is already used in many other places and the signature Api.actual is also a bit nonsensical. I now renamed it to `Signal` with the only required method `.at()`. That matches with our usage as a `grid_signal` in the microgrids